configure

```diff
@@ -68,42 +70,100 @@ while true; do
       ;;
 
     --readline=*)
-      # \( is required for non-GNU expr, which doesn't like leading dashes
-      # See https://www.freebsd.org/cgi/man.cgi?query=expr&sektion=&n=1,
-      # specifically the section on 'Compatibility with previous implementations'
-      FLAG_readline=$(expr \( "$1" : '--readline=\(.*\)' \) )
+      FLAG_readline="${1#*=}"
       ;;
     --readline)
       if test $# -eq 1; then
         die "--readline requires an argument"
       fi
       shift
-      FLAG_readline=$1
+      FLAG_readline="$1"
       ;;
 
     # TODO: Maybe prefix only needs to be part of the install step?  I'm not
     # sure if we need it for building anything.
     --prefix=*)
-      FLAG_prefix=$(expr \( "$1" : '--prefix=\(.*\)' \) )
+      FLAG_prefix="${1#*=}"
       ;;
     --prefix)
       if test $# -eq 1; then
         die "--prefix requires an argument"
       fi
       shift
-      FLAG_prefix=$1
+      FLAG_prefix="$1"
       ;;
 
     # Following autoconf's spelling of --mandir
     --datarootdir=*)
-      FLAG_datarootdir=$(expr \( "$1" : '--datarootdir=\(.*\)' \) )
+      FLAG_datarootdir="${1#*=}"
       ;;
     --datarootdir)
       if test $# -eq 1; then
         die "--datarootdir requires an argument"
       fi
       shift
-      FLAG_datarootdir=$1
+      FLAG_datarootdir="$1"
+      ;;
+
+    --with-*|--enable-*)
+      info "Argument ignored '$1'"
+      ;;
+
+    --build=*)
+      info "Argument ignored '$1'"
+      ;;
+    --build)
+      info "Argument ignored '$1'"
+      shift
+      ;;
+    --host=*)
+      info "Argument ignored '$1'"
+      ;;
+    --host)
+      info "Argument ignored '$1'"
+      shift
+      ;;
+    --mandir=*)
+      info "Argument ignored '$1'"
+      ;;
+    --mandir)
+      info "Argument ignored '$1'"
+      shift
+      ;;
+    --infodir=*)
+      info "Argument ignored '$1'"
+      ;;
+    --infodir)
+      info "Argument ignored '$1'"
+      shift
+      ;;
+    --datadir=*)
+      info "Argument ignored '$1'"
+      ;;
+    --datadir)
+      info "Argument ignored '$1'"
+      shift
+      ;;
+    --sysconfdir=*)
+      info "Argument ignored '$1'"
+      ;;
+    --sysconfdir)
+      info "Argument ignored '$1'"
+      shift
+      ;;
+    --localstatedir=*)
+      info "Argument ignored '$1'"
+      ;;
+    --localstatedir)
```

Comment:
Please document where all these dirs come from, e.g. point to Gentoo or the GNU standards, etc.

Reply:
https://www.gnu.org/prep/standards/html_node/Directory-Variables.html#Directory-Variables
Right above the parse-loop, where you added it. It is still valid.

Reply:
OK well it should have a comment about why we are accepting and ignoring certain flags.

I know you told me but it should be in the code too, for the next person who goes in there.